src/bit_cuda/bit_cuda_transformer.h

```diff
@@ -50,42 +55,50 @@ class BitCudaTransformer : public CfgVisitor<LatticeType>
     }
 
     void visit_functionEntry(FunctionEntryNode<LatticeType>& node) { 
-        node_to_index[&node] = nodes.size();
+        int id = nodes.size();
+        node_to_index[&node] = id;
         bit_cuda::Node& node_struct = nodes.emplace_back();
 
         node_struct.join_mask = 0;
 
         if(node.arguments.size() >= 1){
             node_struct.first_transfer_index = add_transfer_function(node.formal_parameters[0], node.arguments[0]);
+            add_id(id, transfer_functions[node_struct.first_transfer_index].rhs);
             bit_cuda::Transfer& last = transfer_functions[node_struct.first_transfer_index];
             
             for (int i = 1; i < node.arguments.size(); i++)
             {
                 last.next_transfer_index = add_transfer_function(node.formal_parameters[i], node.arguments[i]);
+                add_id(id, transfer_functions[last.next_transfer_index].rhs);
```

Comment:
If the taint constant is used in two seperate arguments and thus get a add_id call. Wouldnt the node id be added to taint_sources twice?

Reply:
changed to store it in a set instead of a vector